sdk/src/test/java/com/horizen/account/transaction/EthereumTransactionTest.java

```diff
@@ -44,66 +45,107 @@ public void BeforeEachTest() throws InvalidAlgorithmParameterException, NoSuchAl
-            exceptionOccurred = true;
+            fail("Test2: Successful EthereumTransaction creation expected.");
         }
-        assertFalse("Test1: Successful EthereumTransaction creation expected.", exceptionOccurred);
 
         // Test 2: raw transaction is null
-        exceptionOccurred = false;
+        boolean exceptionOccurred = false;
         try {
             new EthereumTransaction(null);
         } catch (NullPointerException e) {
             exceptionOccurred = true;
         }
-        assertTrue("Test2: Exception during EthereumTransaction creation expected.", exceptionOccurred);
+        assertTrue("Test3: Exception during EthereumTransaction creation expected.", exceptionOccurred);
 
-        // Test 3: toString function returns correctly
+        // Test 4: toString function returns correctly
         ethereumTransaction = new EthereumTransaction(rawTransaction);
         assertEquals("Ethereum Transaction to String expected to be equal",
-                "EthereumTransaction{from=, nonce=0x1, gasPrice=0x1, gasLimit=0x1, to=0x, data=, Signature=}",
+                "EthereumTransaction{from=, nonce=0x1, gasPrice=0x1, gasLimit=0x1, to=0x, value=0x1, data=0x, " +
+                        "Signature=}",
                 ethereumTransaction.toString());
 
-        // Test 4: ethereum transaction object returns transaction type id correctly
+        // Test 5: ethereum transaction object returns transaction type id correctly
         assertEquals((byte) 2, ethereumTransaction.transactionTypeId());
 
-        // Test 5: ethereum transaction object returns version correctly
-        assertEquals(1, ethereumTransaction.version());
+        // Test 6: ethereum transaction object returns version correctly
+        assertEquals(0/*legacy*/, ethereumTransaction.version());
 
-        // Test 6: ethereum transaction instance returns messageToSign correctly
+        // Test 7: ethereum transaction instance returns messageToSign correctly
         EthereumTransaction ethereumTransactionDeserialize = EthereumTransactionSerializer.getSerializer().parseBytes(ethereumTransaction.bytes());
         assert (Arrays.equals(ethereumTransaction.messageToSign(), ethereumTransactionDeserialize.messageToSign()));
 
-        // Test 7: ethereum transaction instance returns Signature correctly
-        assertEquals(null, ethereumTransaction.getSignature());
+        // Test 7: getMaxFeePerGas should be null, as this is a legacy transaction
+        assertNull(ethereumTransaction.getMaxFeePerGas());
+        // Test 8: getMaxPriorityFeePerGas should be null, as this is a legacy transaction
+        assertNull(ethereumTransaction.getMaxPriorityFeePerGas());
+        // Test 9: getChainId should be null, as this is a legacy transaction
+        assertNull(ethereumTransaction.getChainId());
+        // Test 10: getFrom should be null, as this is an unsigned tx
+        assertNull(ethereumTransaction.getFrom());
```

Comment:
This will always fail now as `getFrom()` returns an empty string and not null anymore, right? :sweat_smile: 
With my comment earlier I wasn't implying that anything was wrong, I was just curious if there was a difference between an empty-string and null. Maybe always use a proper address or null, and don't use empty-string.

Reply:
Oh yes, you're right. Missed that when I fixed it